src/lib.rs

```diff
@@ -398,3 +398,41 @@ pub fn try_cast_slice_mut<
 ) -> Result<&mut [B], PodCastError> {
   unsafe { internal::try_cast_slice_mut(a) }
 }
+
+/// Fill all bytes of `target` with zeroes (see [`Zeroable`]).
+///
+/// This is similar to `*target = Zeroable::zeroed()`, but guarantees that any
+/// padding bytes in `target` are zeroed as well.
+///
+/// See also [`fill_zero`], if you have a slice rather than a single value.
+#[inline]
+pub fn write_zero<T: Zeroable>(target: &mut T) {
+  unsafe {
+    core::ptr::drop_in_place(target);
+    core::ptr::write_bytes(
+      target as *mut T as *mut u8,
+      0u8,
+      core::mem::size_of::<T>(),
+    )
+  }
```

Comment:
This will lead to a double-drop if `drop_in_place` panics. You should probably call `write_byte` in a drop guard instead.

Reply:
This is true. Alternately, do we want to require Copy after all?

Reply:
Alright, just checking i hadn't forgotten anything i guess

Reply:
hmmm, lets just make these require Copy

Reply:
Oh, sure, we can do it in the drop guard.